packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -652,6 +654,20 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
      * @param [options] Options for the request queue
      */
     async run(requests?: (string | Request | RequestOptions)[], options?: CrawlerAddRequestsOptions): Promise<FinalStatistics> {
+        if (this.running) {
+            throw new Error('This crawler instance is already running, you can add more requests to it via `crawler.addRequests()`.');
+        }
+
+        // executing the run method for the second time explicitly, we need to purge the default RQ
+        if (this.running === false && this.requestQueue?.name === 'default') {
+            await this.requestQueue.drop();
+            this.requestQueue = await RequestQueue.open();
+        }
+
+        this.running = true;
```

Comment:
Shouldn't we set `this.running` only when it actually starts running?

Reply:
The idea here was flag it as early as possible, so concurrent calls will result in correct behaviour (=failure). But that was one of my questions from above - maybe it would be better to reuse the state of the autoscaled pool? It should be pretty much the same, right?

Reply:
That's a good question. Frankly I don't know if AutoscaledPool keeps running on the background when the previous `.run()` resolves or if it stops. In any case, I guess this depends on the situation. When you're running the crawlers back to back, we should reuse the state, because it does not make sense to start from zero again, when "we know" that the system can handle higher concurrency. But when you're running the crawlers with some time between the runs, the status of the system could change and therefore the old concurrency might not be ideal anymore...